src/serviceDesks/incontact/inContactServiceDesk.ts

```diff
@@ -0,0 +1,210 @@
+/**
+ * (C) Copyright IBM Corp. 2020.
+ *
+ * Licensed under the MIT License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://opensource.org/licenses/MIT
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ */
+
+import { ErrorType } from '../../types/errors';
+import { MessageRequest, MessageResponse } from '../../types/message';
+import { ServiceDesk, ServiceDeskFactoryParameters, ServiceDeskStateFromWAC } from '../../types/serviceDesk';
+import { AgentProfile, ServiceDeskCallback } from '../../types/serviceDeskCallback';
+import { stringToMessageResponseFormat } from '../../utils';
+import { inContactSession } from './inContactTypes';
+
+class inContactServiceDesk implements ServiceDesk {
+  callback: ServiceDeskCallback;
+
+  state: any;
+
+  /**
+   * Object obtained from a call to the endpoint that is needed for all calls.
+   */
+  session: inContactSession;
+
+  /**
+   * The profile information for the current agent.
+   */
+  agent: AgentProfile = { id: 'liveagent', nickname: 'Live Agent' };
+
+  /**
+   * The object that controls the current polling operation. A new object is created each time polling begins and
+   * that polling can be stopped by setting the value in here to true.
+   */
+  private poller: { stop: boolean };
+
+  /**
+   * The URL of the server that will make requests to Patron API
+   */
+  private SERVER_BASE_URL: string = process.env.SERVER_BASE_URL || 'http://localhost:3000';
+
+  constructor(parameters: ServiceDeskFactoryParameters) {
+    this.callback = parameters.callback;
+  }
+
+  async endChat(): Promise<void> {
```

Comment:
curious to know if inContact Service Desk provides an API where you can leverage it to perform various tasks like sending a message to service desk etc, for ex: https://github.com/watson-developer-cloud/assistant-web-chat-service-desk-starter/blob/main/src/serviceDesks/twilio/twilioFlex.ts#L165 and https://github.com/watson-developer-cloud/assistant-web-chat-service-desk-starter/blob/main/src/serviceDesks/genesys/genesysServiceDesk.ts#L435 instead of implementing it in middleware server

Reply:
nvm, looked through the docs in the Readme, looks like they don't :)